dev/dask_integration/Dockerfile

```diff
@@ -0,0 +1,88 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+FROM ubuntu:14.04
+ADD . /apache-arrow
+WORKDIR /apache-arrow
+# Basic OS utilities
+RUN apt-get update && apt-get install -y \
+        wget \
+        git \
+        gcc \
+        g++
+# This will install conda in /home/ubuntu/miniconda
+RUN wget -O /tmp/miniconda.sh \
+    https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh && \
+    bash /tmp/miniconda.sh -b -p /home/ubuntu/miniconda && \
+    rm /tmp/miniconda.sh
+# Create Conda environment
+ENV PATH="/home/ubuntu/miniconda/bin:${PATH}"
+RUN conda create -y -q -n test-environment \
+    python=3.6
```

Comment:
Is there any opportunity to share a base image with https://github.com/apache/arrow/blob/master/python/testing/hdfs/Dockerfile?

Reply:
I did try to, but the image in that file `cpcloud86/impala:metastore` does not even exist any more! Do you want me to try `cpcloud86/impala:latest` instead?

Reply:
I think `java8-1` is the new latest tag https://github.com/apache/arrow/pull/1031#issuecomment-337780245 cc @cpcloud.

Reply:
Decided in offline discussion to not use the impala dockerfile

Reply:
To relay the contents of this discussion here, the Docker image with Impala on it is very heavyweight at the moment, so we would need to do a fair bit of refactoring to create a a new base image for the HDFS testing image that can be shared with this one. I think it would be useful to do in the future as we add more Docker-based integration testing with downstream Arrow users